src/watts/plugin_dakota.py

```diff
@@ -0,0 +1,268 @@
+# SPDX-FileCopyrightText: 2022 UChicago Argonne, LLC
+# SPDX-License-Identifier: MIT
+
+import os
+import glob
+from datetime import datetime
+from pathlib import Path
+import shutil
+import re
+import pickle
+from typing import List, Optional
+
+import json
+import csv
+import numpy as np
+import pandas as pd
+import subprocess
+
+from .fileutils import PathLike, run as run_proc
+from .parameters import Parameters
+from .plugin import TemplatePlugin
+from .results import Results
+
+
+class ResultsDakota(Results):
+    """Dakota simulation results
+
+    Parameters
+    ----------
+    params
+        Parameters used to generate inputs
+    name
+        Name of workflow producing results
+    time
+        Time at which workflow was run
+    inputs
+        List of input files
+    outputs
+        List of output files
+
+    Attributes
+    ----------
+    stdout
+        Standard output from Dakota run
+    output_data
+        Dictionary with data from .dat files
+    """
+    def __init__(self, params: Parameters, name: str, time: datetime,
+                 inputs: List[PathLike], outputs: List[PathLike]):
+        super().__init__(params, name, time, inputs, outputs)
+        self.output_data = self._get_Dakota_output(params)
+
+    def _get_Dakota_output(self, params: Parameters) -> dict:
+        """Read all Dakota '.dat' files and return results in a dictionary
+
+        Returns
+        -------
+        Results from Dakota .dat files
+
+        """
+
+        if 'dakota_out_file' in params.keys():
+            dakota_out_file_name = params['dakota_out_file']
+        else:
+            dakota_out_file_name = 'dakota_opt.dat'
+
+        # Save Dakota's main output '.dat' files
+        output_data = {}
+        
+        if (glob.glob(dakota_out_file_name)):
+            with open(dakota_out_file_name) as f:
+                reader = csv.reader(f)
+                rows = [row for idx, row in enumerate(reader) if idx == 0]
+      
+            col_names = str(rows[0][0]).split()
```

Comment:
If I'm understanding this correctly, you should be able to do this as:
```suggestion
                col_names = f.readline().split()
```